prov/shm/src/smr_ep.c

```diff
@@ -618,16 +618,16 @@ static int smr_endpoint_name(char *name, char *addr, size_t addrlen,
 			     int dom_idx, int ep_idx)
 {
 	const char *start;
-	memset(name, 0, NAME_MAX);
-	if (!addr || addrlen > NAME_MAX)
+	memset(name, 0, SMR_NAME_MAX);
+	if (!addr || addrlen > SMR_NAME_MAX)
 		return -FI_EINVAL;
 
 	start = smr_no_prefix((const char *) addr);
 	if (strstr(addr, SMR_PREFIX) || dom_idx || ep_idx)
-		snprintf(name, NAME_MAX, "%s:%d:%d:%d", start, getuid(), dom_idx,
-			 ep_idx);
+		snprintf(name, SMR_NAME_MAX, "%s:%d:%d:%d", start, getuid(),
+			 dom_idx, ep_idx);
 	else
-		snprintf(name, NAME_MAX, "%s", start);
+		snprintf(name, SMR_NAME_MAX, "%s", start);
```

Comment:
Why do the other snprintf calls use max - 1?

Reply:
Ugh because this one is wrong. Please hold.